exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/PruneScanRule.java

```diff
@@ -520,4 +531,10 @@ private static boolean isQualifiedDirPruning(final TableScan scan) {
     return false;
   }
 
+  private static void setNotPruned(MetadataContext metaContext) {
```

Comment:
What about setPruneStatus(MetadataContext metaContext, PruneStatus status)? Then, both setNotPruned() and code on line 437-440 could share this call.